src/metabase/sync_database/analyze.clj

```diff
@@ -157,86 +53,42 @@
-        (assoc field-stats :special-type :type/Email, :preview-display true)))))
-
-(defn- test:new-field
-  "Do the various tests that should only be done for a new `Field`.
-   We only run most of the field analysis work when the field is NEW in order to favor performance of the sync process."
-  [driver field field-stats]
-  (->> field-stats
-       (test:no-preview-display driver field)
-       (test:url-special-type   driver field)
-       (test:json-special-type  driver field)
-       (test:email-special-type driver field)))
-
-;; TODO - It's weird that this one function requires other functions as args when the whole rest of the Metabase driver system
-;;        is built around protocols and record types. These functions should be put back in the `IDriver` protocol (where they
-;;        were originally) or in a special `IAnalyzeTable` protocol).
-(defn make-analyze-table
-  "Make a generic implementation of `analyze-table`."
-  {:style/indent 1}
-  [driver & {:keys [field-avg-length-fn field-percent-urls-fn calculate-row-count?]
-             :or   {field-avg-length-fn   (partial driver/default-field-avg-length driver)
-                    field-percent-urls-fn (partial driver/default-field-percent-urls driver)
-                    calculate-row-count?  true}}]
-  (fn [driver table new-field-ids]
-    (let [driver (assoc driver :field-avg-length field-avg-length-fn, :field-percent-urls field-percent-urls-fn)]
-      {:row_count (when calculate-row-count? (u/try-apply table-row-count table))
-       :fields    (for [{:keys [id] :as field} (table/fields table)]
-                    (let [new-field? (contains? new-field-ids id)]
-                      (cond->> {:id id}
-                               (test-for-cardinality? field new-field?) (test:cardinality-and-extract-field-values field)
-                               new-field?                               (test:new-field driver field))))})))
-
-(defn generic-analyze-table
-  "An implementation of `analyze-table` using the defaults (`default-field-avg-length` and `field-percent-urls`)."
-  [driver table new-field-ids]
-  ((make-analyze-table driver) driver table new-field-ids))
+(defn field-fingerprint [driver table field]
+  (let [values (db/select 'FieldValues :id (:id table))
+        #_values-to-test-for-json-and-email #_(take driver/max-sync-lazy-seq-results 
+                                                    (driver/field-values-lazy-seq driver field))]
+      {:id                                (:id field) ;; check if this should be field or table id
+       :field-percent-urls      (u/try-apply (:field-percent-urls driver) field)
+       :field-percent-json      (if (values-are-valid-json? values) 100 0)
+       :field-percent-email     (if (values-are-valid-emails? values) 100 0)
+       :field-avg-length        (u/try-apply (:field-avg-length driver) field)
+       :visibility_type         (:visibility_type field)
+       :values                  values
```

Comment:
Values should not be in the fingerprint. They will be both large and potentially sensitive.